### PR TITLE
build: update node packages

### PR DESCRIPTION
**Pull request by cybervoid0**
# What

- upgrade the main packages (hardhat, ethers, linters, formatters)
- fix an issue where eslint didnt show prettier warnings/errors (install eslint-plugin-prettier)

# Why

to fix eslint issue + for keeping project up-to-date and safe

### CODE REVIEW

File: hardhat.config.ts [hunk 1]
```diff
@@ -4,7 +4,15 @@ import '@openzeppelin/hardhat-upgrades'
 
 const config: HardhatUserConfig = {
   solidity: {
-    compilers: [{ version: '0.8.24' }, { version: '0.4.24' }],
+    compilers: [
+      {
+        version: '0.8.20',
```
Comment (TravellerOnTheRun) [inline hardhat.config.ts:9]: Why 0.8.20? My Governor is using 0.8.24, is there any specific reason to use older version

**Review by TravellerOnTheRun — CHANGES_REQUESTED** [review]

File: hardhat.config.ts (hunk 1 above)
Comment (cybervoid0) [reply hardhat.config.ts:9]: yes there is! Slither believes this version in unstable

Comment (cybervoid0) [reply hardhat.config.ts:9]: INFO:Detectors:
Version constraint ^0.8.24 contains known severe issues (https://solidity.readthedocs.io/en/latest/bugs.html)
.
It is used by:
        - ^0.8.24 (contracts/StRIFToken.sol#3)
Reference: https://github.com/crytic/slither/wiki/Detector-Documentation#incorrect-versions-of-solidity

**Review by Freshenext — APPROVED** [review]
Looks good!

**Review by rodrigoncalves — APPROVED** [review]

**Review by TravellerOnTheRun — APPROVED** [review]

**Review by sleyter93 — APPROVED** [review]
Good Job!

**Comment by cybervoid0** [comment]
> Good Job!

No job here yet :) Just up and down grades. Job will be in the next PR

**Pull request merged**